it/v2/connectors/PersistenceConnectorSpec.scala

```diff
@@ -921,9 +921,108 @@ class PersistenceConnectorSpec
     }
   }
 
-  private def generateDepartureResponse(departureId: MovementId) =
-    DepartureResponse(
-      _id = departureId,
+  "GET /traders/:EORI/movements/arrivals" - {
+
+    implicit val hc = HeaderCarrier()
+    val eori        = arbitrary[EORINumber].sample.get
+
+    def targetUrl(eoriNumber: EORINumber) = s"/transit-movements/traders/${eoriNumber.value}/movements/arrivals"
+
+    "on success, return a list of arrival IDs" in {
+      lazy val arrivalIdList = Gen.listOfN(3, arbitrary[MovementId]).sample.get
+      lazy val messageResponse = arrivalIdList.map(
+        id => generateMovementResponse(id)
+      )
+
+      server.stubFor(
+        get(
+          urlEqualTo(targetUrl(eori))
+        )
+          .willReturn(
+            aResponse()
+              .withStatus(OK)
+              .withBody(
+                Json.toJson(messageResponse).toString()
+              )
+          )
+      )
+
+      implicit val hc = HeaderCarrier()
+      val result      = persistenceConnector.getArrivalsForEori(eori)
+      whenReady(result) {
+        _ mustBe messageResponse
+      }
+    }
+
+    "on incorrect Json, return an error" in {
+      server.stubFor(
+        get(
+          urlEqualTo(targetUrl(eori))
+        )
+          .willReturn(
+            aResponse()
+              .withStatus(OK)
+              .withBody(
+                "{ \"test\": \"fail\" }"
+              )
+          )
+      )
+
+      val result = persistenceConnector
+        .getArrivalsForEori(eori)
+        .map(
+          _ => fail("This should have failed with a JsResult.Exception, but it succeeded")
+        )
+        .recover {
+          case JsResult.Exception(_)  => ()
+          case t: TestFailedException => t
+          case thr                    => fail(s"Expected a JsResult.Exception, got $thr")
+        }
+
+      whenReady(result) {
+        _ mustBe (())
+      }
+    }
+
+    "on an internal error, return an UpstreamServerError" in {
+      server.stubFor(
+        get(
+          urlEqualTo(targetUrl(eori))
+        )
+          .willReturn(
+            aResponse()
+              .withStatus(INTERNAL_SERVER_ERROR)
+              .withBody(
+                Json.stringify(
+                  Json.obj(
+                    "code"    -> "INTERNAL_SERVER_ERROR",
+                    "message" -> "Internal server error"
+                  )
+                )
+              )
+          )
+      )
+
+      val result = persistenceConnector
+        .getArrivalsForEori(eori)
+        .map(
+          _ => fail("This should have failed with an UpstreamErrorResponse, but it succeeded")
+        )
+        .recover {
+          case UpstreamErrorResponse(_, status, _, _) => status
+          case thr                                    => fail(s"Expected an UpstreamErrorResponse with a 500, got $thr")
+        }
+
+      whenReady(result) {
+        _ mustBe INTERNAL_SERVER_ERROR
+      }
+    }
+
+  }
```

Comment:
please add test cases to cover 404 not found

Reply:
404 is handled by the service

Reply:
I think we can handle 404 here as well. Because from transit-movements we are handing both type of error 500 and 404. I am ok if not required